From 19544c0d2c14051e0cea65478b6106af76b836b6 Mon Sep 17 00:00:00 2001
From: Mukesh Ojha <mukesh.ojha@oss.qualcomm.com>
Date: Tue, 27 Jan 2026 14:01:00 +0530
Subject: [PATCH 1/2] FROMLIST: arm64: dts: qcom: lemans: disable zap-shader
 for EL2 configuration

We don't need to use zap shader in EL2 as Linux can zap the gpu on
it's own. Lets disable zap-shader for Lemans EL2 configuration.

Change-Id: I0f4dfe93cf49e82896e074a77e65a682708fcc15
Upstream-Status: Submitted [https://lore.kernel.org/all/20260127-talos-el2-overlay-v2-1-b6a2266532c4@oss.qualcomm.com/]
Signed-off-by: Mukesh Ojha <mukesh.ojha@oss.qualcomm.com>
---
 arch/arm64/boot/dts/qcom/sa8775p-el2.dtso | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sa8775p-el2.dtso b/arch/arm64/boot/dts/qcom/sa8775p-el2.dtso
index 0d2c9302bda6..bd5e5de79249 100644
--- a/arch/arm64/boot/dts/qcom/sa8775p-el2.dtso
+++ b/arch/arm64/boot/dts/qcom/sa8775p-el2.dtso
@@ -26,6 +26,10 @@ &remoteproc_cdsp1 {
 	iommus = <&apps_smmu 0x29c0 0x0400>;
 };
 
+&gpu_zap_shader {
+	status = "disabled";
+};
+
 &iris {
 	video-firmware {
 		iommus = <&apps_smmu 0x0882 0x0400>;
-- 
2.50.1

